### ISSUE TITLE
Namespace issues with prefixes reuse in twisted.words.xish.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2609]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)| @ralphm reported|
|-|-|
|Trac ID|trac#2609|
|Type|enhancement|
|Created|2007-04-27 15:05:10Z|

Alerted by Florian Holthauer, a namespace issue occurs by repeatedly passing the same dictionary for prefixes to an `Element`'s `toXml` method. The serialization causes the passed dictionary to modified, causing havoc for subsequent use of namespaces later on.

<details><summary>Searchable metadata</summary>

```
trac-id__2609 2609
type__enhancement enhancement
reporter__ralphm ralphm
priority__highest highest
milestone__Words_0_6 Words-0.6
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1177686310000000 1177686310000000
changetime__1177837126000000 1177837126000000
version__None None
owner__ 
cc__ralphm cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_1"></a>|@ralphm commented|
|-|-|

The passed dictionary not copied but modified in place. Created branch [branches/xish-prefixes-reuse-2609,](branches/xish-prefixes-reuse-2609,) added test and fix.

Please review

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)|@ralphm removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @ralphm|
|-|-|

 * test_namespaceReuse and test_prefixesReuse need a docstring
 * in test_prefixesReuse, you could probably test explicitely the value of the prefixes dict to check it's unchanged by the Element call
 * the from !__future!__ import can be removed

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_5"></a>|@ralphm set owner to @therve|
|-|-|

Thanks for the comments.

I added the explicit test for the prefixes parameter, and addressed the other remarks.

Please review again.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve set owner to @ralphm|
|-|-|

Looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_7"></a>|@ralphm set status to `closed`|
|-|-|

(In [20101]) Fix reuse of dict of prefixes passed to t.w.x.domish.Element.toXml.

Author: ralphm
Reviewer: therve
Fixes [#2609](https://github.com/twisted/twisted/issues/2609).

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
